02-microarray/pathway-analysis_microarray_03_gsea.Rmd

```diff
@@ -333,15 +336,169 @@ Looks like we were able to successfully get rid of the duplicate gene identifier
 
 ## Perform gene set enrichment analysis (GSEA)
 
-_Addressed in upcoming PR_
+The goal of GSEA is to determine whether the members of a set of genes are randomly distributed throughout a pre-ranked gene list or primarily found at the top or bottom of the list.
+As mentioned earlier, hallmark gene sets summarize and represent specific well-defined biological states or processes and display coherent expression of genes.
+We therefore expect that sets will tend to be highly- or lowly-ranked or, put another way, found at the top or bottom of the pre-ranked gene list [@clusterProfiler-book].
+
+### Determine our pre-ranked genes list
+
+The `GSEA()` function takes a pre-ranked and sorted named vector of statistics, where the names in the vector are gene identifiers.
+This is step 1 -- create a gene list with statistics that GSEA will rank by.
+
+```{r}
+# Let's create a named vector ranked based on the t-statistic values -- we
+# need to sort the t-statistic values in descending order here
+t_vector <- filtered_dge_mapped_df$t
+names(t_vector) <- filtered_dge_mapped_df$entrez_id
+t_vector <- sort(t_vector, decreasing = TRUE)
```

Comment:
```suggestion
# We need to sort the t-statistic values in descending order here
t_vector <- sort(t_vector, decreasing = TRUE)
```
And then delete the second half of the comment above. I think placing the comment with its corresponding code makes it easier for the user to parse out which step is which.